### DISCUSSION TITLE
Login to Torus wallet with web3auth user info

### DISCUSSION THREAD

**Discussion by seakkus** [Web3Auth/Web3Auth#829] [Help]
Hello,

Is there a way to login/show the torus wallet once the user is logged with web3auth? I am using firebase custom auth with web3auth. I am able to login users and get their user info (address, private key etc) but I am not able to log them into Torus wallet or any of the wallet UI provided by web3auth/torus. Is there any way?

Here is what I have tried but failed so far:

`
  const userInfo = await web3auth.getUserInfo();
  const privateKey: string = await web3auth.provider.request({
          method: "eth_private_key"
      });

    const torus = new Torus();
    await torus.init({
        buildEnv: 'testing',
        enableLogging: false,
        showTorusButton: false,
        network: {
            host: 'mumbai',
            chainId: 80001,
            networkName: 'Polygon Mumbai'
        }
    });
    await torus.loginWithPrivateKey({ privateKey, userInfo });
    await torus.showWallet("home");`

The error I am getting is PreferencesController.js:389 unable to fetch etherscan tx Response {type: 'cors', url: 'https://api.tor.us/etherscan?selectedAddress=0x8fc…18816d374462090a416ab216b&selectedNetwork=mainnet

**Comment by shahbaz17** [comment]
Hey @seakkus 

Do you want the TorusWallet Plugin Widget on your dApp? If so, please take a look at https://web3auth.io/docs/sdk/web/plugins/torus-wallet#example.

```js
import { TorusWalletConnectorPlugin } from "@web3auth/torus-wallet-connector-plugin";

const torusPlugin = new TorusWalletConnectorPlugin({
  torusWalletOpts: {},
  walletInitOptions: {
    whiteLabel: {
      theme: { isDark: true, colors: { primary: "#00a8ff" } },
      logoDark: "https://web3auth.io/images/w3a-L-Favicon-1.svg",
      logoLight: "https://web3auth.io/images/w3a-D-Favicon-1.svg",
    },
    useWalletConnect: true,
    enableLogging: true,
  },
});
await web3auth.addPlugin(torusPlugin);
```
